pom.xml

```diff
@@ -342,6 +345,49 @@
           <artifactId>nexus-staging-maven-plugin</artifactId>
           <version>${version.nexus.staging.plugin}</version>
         </plugin>
+        <plugin>
+          <groupId>net.revelc.code.formatter</groupId>
+          <artifactId>formatter-maven-plugin</artifactId>
+          <version>${version.formatter.plugin}</version>
+            <dependencies>
+              <dependency>
+                <groupId>io.smallrye</groupId>
+                <artifactId>smallrye-code-rules</artifactId>
+                <version>${version.smallrye.code.rules.plugin}</version>
+              </dependency>
+            </dependencies>
+            <configuration>
+              <configFile>io/smallrye/coderules/eclipse-format.xml</configFile>
+              <skip>${format.skip}</skip>
+            </configuration>
+            <executions>
+              <execution>
+                <phase>process-sources</phase>
+                <goals>
+                  <goal>format</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>net.revelc.code</groupId>
+            <artifactId>impsort-maven-plugin</artifactId>
+            <version>${version.impsort.plugin}</version>
+            <configuration>
+              <groups>java.,javax.,org.,com.</groups>
+              <staticGroups>*</staticGroups>
+              <skip>${format.skip}</skip>
+              <removeUnused>true</removeUnused>
+            </configuration>
+            <executions>
+              <execution>
+                <id>sort-imports</id>
+                <goals>
+                  <goal>sort</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
       </plugins>
```

Comment:
Would it be beneficial to have the profiles present in here as well?

Then when a project updates to the latest parent they pick up the formatting without having to add it directly?

Reply:
This is included in the ".*" rule. that says "all files starting with are dot should be ignored".
I think this is also more secure, as normally such files should never be in source control as they often also may contain secrets.

```
.*
!/.gitignore
!*.gitkeep
!.mvn
target
*.log
```

Reply:
> Would it be beneficial to have the profiles present in here as well?
What profiles do you mean?

> Then when a project updates to the latest parent they pick up the formatting without having to add it directly?
I think this should be up to the projects. Formatting everything automatically may cause merge problems for open work. I think they should try to close all open pulls first and then format the code once. This way every project has the chance to decide when to do it. Later on this could be included by default when every project is up to date.

Reply:
Didn't notice the `,*`.

I mean profiles similar to Quarkus for the formatting and validation.

It would be nice if it didn't need to be duplicated across all projects. I agree it would be nice to let the projects decide when to implement the formatting, but they can do that by choosing when to update the parent.

I fear requiring the projects to include the profiles/plugins themselves will just mean it never gets done

Reply:
I looked at the profiles at quarkus and didn't really get the sense of why to use profiles here. Seems to be very complicated and they are included by default. I think you could achieve the same by simply adding the plugins to the "plugin" instead of "pluginmanagement" section. If you want to disable them, there is the "format.skip" property. The only thing I'm not sure how this affects non-java projects ("pom" or other types without java code).

Reply:
I'm fine with just adding the plugins directly into the `plugins` section of the pom.

I don't believe it affects them, as there are pom modules in Quarkus that inherit from it's build-parent

Reply:
Ok, just done the same thing :-)